src/azul/plugins/__init__.py

```diff
@@ -428,15 +428,22 @@ def special_fields(self) -> SpecialFields:
         """
         raise NotImplementedError
 
+    @property
+    def root_entity_type(self) -> str:
+        """
+        The type of entity that sits at the root of the entity graph, containing
+        entities of all other types.
+        """
+        raise NotImplementedError
+
     @property
     @abstractmethod
-    def implicit_hub_type(self) -> str:
+    def hot_entity_types(self) -> Iterable[str]:
         """
-        The type of entities that do not explicitly track their hubs in replica
-        documents in order to avoid a large list of hub references in the
-        replica document, and to avoid contention when updating that list during
-        indexing. Note that this is not a type of hub entities, but rather the
-        type of replica entities that have implicit hubs.
+        The types of inner entities that do not explicitly track their hubs in
+        replica documents in order to avoid a large list of hub references in
+        the replica document, and to avoid contention when updating that list
+        during indexing. This should always include the root type.
```

Comment:
Are there any checks of that requirement, or other means of enforcing it?

Reply:
I've added an assertion